app/javascript/client/home/utils.tsx

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from 'react';
+
+interface size {
+    width: number,
+    height: number
+}
+
+export const useWindowSize = ()=>{
+
+    const [windowSize, setWindowSize] = useState({
+      width: window.innerWidth,
+      height: window.innerHeight,
+    });
+  
+    useEffect(() => {
+      function handleResize() {
+        setWindowSize({
+          width: window.innerWidth,
+          height: window.innerHeight,
+        });
+      }
+      
+      window.addEventListener("resize", handleResize);
+      
+      handleResize();
+      
+      return () => window.removeEventListener("resize", handleResize);
+    }, []);
+  
+    return windowSize;
```

Comment:
I think I understand the basic idea of this component, but I definitely don't understand all the details ... would be great to go over it